Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/CategorySample.cs

```diff
@@ -0,0 +1,258 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Runtime.InteropServices;
+
+    internal class CategorySample
+    {
+        public Dictionary<int, CounterDefinitionSample> CounterTable;
+
+        public Dictionary<string, int> InstanceNameTable;
+
+        private readonly PerfLib library;
+
+        public CategorySample(byte[] data, int categoryNameIndex, int counterNameIndex, PerfLib library)
```

Comment:
Add a file-level comment section that explains which file it was copied from and what version of FW you copied it from down to the file version of the file you copied it from.